src/uct/ib/rc/accel/gga_mlx5.c

```diff
@@ -185,17 +185,113 @@ uct_gga_mlx5_rkey_resolve(uct_ib_mlx5_md_t *md, uct_rkey_t rkey)
+    .iface_event_arm          = ucs_empty_function_return_unsupported,
+    .iface_close              = uct_gga_mlx5_iface_t_delete,
+    .iface_query              = (uct_iface_query_func_t)ucs_empty_function_do_assert,
+    .iface_get_address        = ucs_empty_function_return_success,
+    .iface_get_device_address = ucs_empty_function_return_unsupported,
+    .iface_is_reachable       = uct_base_iface_is_reachable
+};
+
+static uct_rc_iface_ops_t uct_gga_mlx5_iface_ops = {
+    .super = {
+        .super = {
+            .iface_estimate_perf   = uct_base_iface_estimate_perf,
+            .iface_vfs_refresh     = uct_rc_iface_vfs_refresh,
+            .ep_query              = (uct_ep_query_func_t)ucs_empty_function,
+            .ep_invalidate         = uct_rc_mlx5_base_ep_invalidate,
+            .ep_connect_to_ep_v2   = (uct_ep_connect_to_ep_v2_func_t)ucs_empty_function_do_assert,
+            .iface_is_reachable_v2 = ucs_empty_function_do_assert,
+            .ep_is_connected       = ucs_empty_function_do_assert
+        },
+        .create_cq      = uct_rc_mlx5_iface_common_create_cq,
+        .destroy_cq     = uct_rc_mlx5_iface_common_destroy_cq,
+        .event_cq       = uct_rc_mlx5_iface_common_event_cq,
+        .handle_failure = (uct_ib_iface_handle_failure_func_t)ucs_empty_function_do_assert_void,
+    },
+    .init_rx         = (uct_rc_iface_init_rx_func_t)ucs_empty_function_do_assert,
+    .cleanup_rx      = ucs_empty_function_do_assert_void,
+    .fc_ctrl         = uct_rc_mlx5_base_ep_fc_ctrl,
+    .fc_handler      = uct_rc_iface_fc_handler,
+    .cleanup_qp      = ucs_empty_function_do_assert_void,
+    .ep_post_check   = uct_rc_mlx5_base_ep_post_check,
+    .ep_vfs_populate = uct_rc_mlx5_base_ep_vfs_populate
+};
+
 static UCS_CLASS_INIT_FUNC(uct_gga_mlx5_iface_t,
                            uct_md_h tl_md, uct_worker_h worker,
                            const uct_iface_params_t *params,
                            const uct_iface_config_t *tl_config)
 {
-    return UCS_ERR_NOT_IMPLEMENTED;
+    uct_gga_mlx5_iface_config_t *config =
+            ucs_derived_of(tl_config, uct_gga_mlx5_iface_config_t);
+    uct_ib_mlx5_md_t *md                = ucs_derived_of(tl_md, uct_ib_mlx5_md_t);
+    uct_ib_iface_init_attr_t init_attr  = {};
+
+    init_attr.qp_type               = IBV_QPT_RC;
+    init_attr.cq_len[UCT_IB_DIR_TX] = config->super.tx_cq_len;
+    init_attr.max_rd_atomic         = IBV_DEV_ATTR(&md->super.dev,
+                                                   max_qp_rd_atom);
+
+    UCS_CLASS_CALL_SUPER_INIT(uct_rc_mlx5_iface_common_t,
+                              &uct_gga_mlx5_iface_tl_ops,
+                              &uct_gga_mlx5_iface_ops, tl_md, worker, params,
+                              &config->super.super, &config->rc_mlx5_common,
+                              &init_attr);
+
+    self->super.super.config.tx_moderation =
+            ucs_min(config->super.tx_cq_moderation, self->super.tx.bb_max / 4);
```

Comment:
maybe we can unite common code with uct_rc_mlx5_iface_t constructor: move these lines to uct_rc_mlx5_iface_common_t, and add a "tx_moderation" field to uct_ib_iface_init_attr_t; for rc/gga it will be config->super.tx_cq_moderation and for dc it will be 0